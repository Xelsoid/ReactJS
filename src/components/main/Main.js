import React from 'react';
import { Switch, Route, Redirect } from "react-router-dom";

import { PATHS } from '../../helpers/constants';
import SortResults from '../sortResults/SortResults';
import FilmsGallery from '../filmsGallery/FilmsGallery';
import FilmDescription from '../filmDescription/FilmDescription';
import NotFound from '../notFound/NotFound';

const Main = () => (
  <main>
    <Switch>
      <Route exact path={PATHS.DEFAULT} component={null} />
      <Route path={`${PATHS.MOVIE}/:name`} component={FilmDescription} />
      <Route
        path={PATHS.MOVIES}
        render={() => (
          <div>
            <SortResults />
            <FilmsGallery />
          </div>
        )}
      />
      <Route path={PATHS.NOTFOUND} component={NotFound} />
      <Redirect from={PATHS.RESTURLS} to={PATHS.NOTFOUND} />
    </Switch>
  </main>
);

export default Main;
