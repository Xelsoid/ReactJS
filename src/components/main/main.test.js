import React from 'react';
import Main from './Main';
import DATA from '../../mockedData/MOCKED_DATA';

const requestedFilms = DATA;

describe('Main', () => {
  it('renders correctly without film description', () => {
    const isDisplayFilmDescription = false;
    const component = shallow(
      <Main requestedFilms={requestedFilms} isDisplayFilmDescription={isDisplayFilmDescription} />
    );
    expect(component).toMatchSnapshot();
  });

  it('renders correctly with film description', () => {
    const isDisplayFilmDescription = true;
    const component = shallow(
      <Main requestedFilms={requestedFilms} isDisplayFilmDescription={isDisplayFilmDescription} />
    );
    expect(component).toMatchSnapshot();
  });
});
