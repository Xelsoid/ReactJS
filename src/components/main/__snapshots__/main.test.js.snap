// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main renders correctly with film description 1`] = `
<Switch>
  <Route
    component={null}
    exact={true}
    path="/"
  />
  <Route
    component={[Function]}
    path="/description/:name"
  />
  <Route
    path="/movies"
    render={[Function]}
  />
  <Route
    component={[Function]}
    path="/page-not-found"
  />
  <Redirect
    from="*"
    to="/page-not-found"
  />
</Switch>
`;

exports[`Main renders correctly without film description 1`] = `
<Switch>
  <Route
    component={null}
    exact={true}
    path="/"
  />
  <Route
    component={[Function]}
    path="/description/:name"
  />
  <Route
    path="/movies"
    render={[Function]}
  />
  <Route
    component={[Function]}
    path="/page-not-found"
  />
  <Redirect
    from="*"
    to="/page-not-found"
  />
</Switch>
`;
